judge/bridge/judgecallback.py

```diff
@@ -178,6 +178,8 @@ def on_grading_end(self, packet):
         submission.case_total = total
 
         sub_points = round(points / total * submission.problem.points if total > 0 else 0, 3)
+        if submission.user.is_unlisted:
```

Comment:
I agree with this. I think the best implementation would allow such users to gain points, but exclude them from all rankings. This is a bit tricky, since there are a lot of places that enumerate users. Off the top of my head;

* users list
* organization users list
* API
* contest ranking list

While these changes are being made, it would also be a good idea to filter out people who have `is_active` toggled off &mdash; we don't want banned users to appear in any form of rankings.